### PR TITLE
Use the 'CONFETTI_PUBLIC_' prefix for exported env vars

### PR DESCRIPTION
**Pull request by axbarsan**
Fixes https://github.com/giantswarm/giantswarm/issues/14958

Due to how deployment works (don't @ me), in production we have some wild env vars, such as:
- `CONFETTI_FRONTEND_PORT_3000_TCP_ADDR`
- `CONFETTI_FRONTEND_PORT_3000_TCP_PORT`
- etc.

Because previously we exported to the window object on the browser side all environmental variables that had the `CONFETTI_` prefix, these variables also got exported, creating a security flaw.

With this PR, we export to the window object on the browser side the env variables that have the `CONFETTI_PUBLIC_` prefix, so no more nasty variables exported.

### CODE REVIEW

File: helm/confetti-frontend/templates/deployment.yaml [hunk 1]
```diff
@@ -24,17 +24,17 @@ spec:
...
+          - name: CONFETTI_PUBLIC_BACKEND_HOST_SECURE
             value: {{ .Values.backendUrlSecure | quote }}
           ports:
             - containerPort: {{ .Values.port }}
               name: http
           resources:
             requests:
               cpu: 100m
-              memory: 20Mi
+              memory: 250Mi
```
Comment (axbarsan) [inline helm/confetti-frontend/templates/deployment.yaml:37]: Taking advantage of this PR. This was seen here https://gigantic.slack.com/archives/CQ465CFG8/p1610382503000900

File: helm/confetti-frontend/templates/ingress.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-apiVersion: extensions/v1beta1
+apiVersion: networking.k8s.io/v1beta1
```
Comment (axbarsan) [inline helm/confetti-frontend/templates/ingress.yaml:1]: Same here

**Review by stone-z — APPROVED** [review]
Seems reasonable to me, thanks!

**Review by oponder — COMMENTED** [review]
LGTM ✅

**Comment by oponder** [comment]
oops 🐌

**Pull request merged**